### PR TITLE
Add `onIncidentUpdate` trigger and function

### LINKED ISSUE #1142: Move Email subscription processing to a Atlas trigger
Move the sending email process for subscriptions to a [Atlas trigger](https://www.mongodb.com/docs/atlas/app-services/triggers/overview/) 

For example, when an incident is updated we should configure an Atlas trigger to send all Incident update email notifications to subscribers, instead of sending them through the app itself.

### PR DESCRIPTION
**Pull request by pdcp1**
Closes #1142

## Testing

Preview link for testing: https://deploy-preview-8--pablocosta-staging.netlify.app

1. Login and subscribe to an incident. ie: https://deploy-preview-8--pablocosta-staging.netlify.app/cite/11
2. Login on another browser with another Admin user
3. Go to https://deploy-preview-8--pablocosta-staging.netlify.app/apps/submitted and promote a report to the incident selected in step # 1
4. The new trigger should be executed and the user in step # 1 should receive the Incident update email notification

### CODE REVIEW

**Comment by pdcp1** [comment]
@cesarvarela this is ready to review. The testing link and testing steps are in the PR description

File: site/realm/data_sources/mongodb-atlas/config.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "type": "mongodb-atlas",
     "config": {
         "clusterName": "AIIDDev",
+        "namespacePreimageConfigs": [],
```
Comment (cesarvarela) [inline site/realm/data_sources/mongodb-atlas/config.json:6]: image

What's this for?

**Review by cesarvarela — COMMENTED** [review]
A couple of questions before mergin

File: site/realm/functions/onIncidentUpdate.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const updatedFields = Object.keys(updateDescription.updatedFields);
+
+  const isReportListChanged = updatedFields.some(field => field.match(/reports/));
+
+  if (isReportListChanged) {
+    const { reports: newReports } = fullDocument;
+    const { reports: oldReports } = fullDocumentBeforeChange;
+
+    // If a new report is added, send the email notification to all subscribers
+    if (newReports.length > oldReports.length) {
```
[the remark this answers is not in the document]
Comment (pdcp1) [reply site/realm/functions/onIncidentUpdate.js:16]: I think if we still allow linking just one report to an incident at a time, this code will work.
Keep in mind that this function will be executed each time an incident document is changed on the DB.
If the new feature allows users to link a report to many incidents, this function will be executed one time for each incident updated.

**Comment by cesarvarela** [comment]
Looks like this was already merged

**Pull request closed without merging**